pqiv.c fails to compile in c23 mode: pqiv.c:4246:62: error: passing argument 2 of ‘g_timeout_add’ from incompatible pointer type

Full error text:

```
pqiv.c:4246:62: error: passing argument 2 of ‘g_timeout_add’ from incompatible pointer type [-Wincompatible-pointer-types]
 4246 |         fullscreen_transition_source_id = g_timeout_add(500, window_fullscreen_helper_reset_transition_id, NULL);
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                              |
      |                                                              gboolean (*)(void) {aka int (*)(void)}
/usr/include/glib-2.0/glib/gmain.h:878:54: note: expected ‘GSourceFunc’ {aka ‘int (*)(void *)’} but argument is of type ‘gboolean (*)(void)’ {aka ‘int (*)(void)’}
  878 |                                      GSourceFunc     function,
      |                                      ~~~~~~~~~~~~~~~~^~~~~~~~
/home/nho/workshop/pqiv/pqiv.c:4162:10: note: ‘window_fullscreen_helper_reset_transition_id’ declared here
 4162 | gboolean window_fullscreen_helper_reset_transition_id() {/*{{{*/
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gmain.h:199:20: note: ‘GSourceFunc’ declared here
  199 | typedef gboolean (*GSourceFunc)       (gpointer user_data);
      |                    ^~~~~~~~~~~
```
Fix: change definition to
`window_fullscreen_helper_reset_transition_id(void *)`
Does exactly same thing, but conforms to standards.

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -167,7 +167,7 @@ ifndef VERBOSE
 endif
 
 # Assemble final compiler flags
-CFLAGS_REAL=-std=gnu99 $(PQIV_WARNING_FLAGS) $(PQIV_VERSION_FLAG) $(CFLAGS) $(DEBUG_CFLAGS) $(EXTRA_DEFS) $(shell $(PKG_CONFIG) --cflags "$(LIBS)")
+CFLAGS_REAL=$(PQIV_WARNING_FLAGS) $(PQIV_VERSION_FLAG) $(CFLAGS) $(DEBUG_CFLAGS) $(EXTRA_DEFS) $(shell $(PKG_CONFIG) --cflags "$(LIBS)")
 LDLIBS_REAL=$(shell $(PKG_CONFIG) --libs "$(LIBS)") $(LDLIBS)
 LDFLAGS_REAL=$(LDFLAGS)
 

--- a/pqiv.c
+++ b/pqiv.c
@@ -4159,7 +4159,7 @@ void do_jump_dialog() { /* {{{ */
 #endif
 // }}}
 /* Main window functions {{{ */
-gboolean window_fullscreen_helper_reset_transition_id() {/*{{{*/
+gboolean window_fullscreen_helper_reset_transition_id(void *) {/*{{{*/
 	action_done();
 	fullscreen_transition_source_id = -1;
 	return FALSE;

